data_structures/src/chain.rs

```diff
@@ -2913,7 +2913,18 @@ impl ReputationEngine {
         self.ars
             .active_identities()
             .cloned()
-            .sorted_by_key(|&pkh| std::cmp::Reverse((self.trs.get(&pkh), pkh)))
+            .sorted_by_key(|&pkh| {
+                let [a0, a1, a2, a3] = self.current_alpha.0.to_be_bytes();
+                let pkh_mod_bytes = [
+                    pkh.hash[0] + a0,
+                    pkh.hash[1] + a1,
+                    pkh.hash[2] + a2,
+                    pkh.hash[3] + a3,
+                ];
+                let hash = Hash::from(calculate_sha256(&pkh_mod_bytes));
```

Comment:
It may be expensive to calculate a SHA256 hash every time we compare two elements, but since it is only used as a tie breaker we can optimize this.

You can use `sorted_by(|a, b| { ...` which takes two elements as input and returns an `Ordering`: `FnMut(&T, &T) -> Ordering`. Then compare the reputation of a with the reputation of b and only calculate the hash when the reputation is equal.

https://doc.rust-lang.org/stable/std/primitive.slice.html#method.sort_by

Reply:
And add tests that show that the ordering changes when `current_alpha` changes.